upgrade_test.py

```diff
@@ -694,44 +698,48 @@ def test_generic_cluster_upgrade(self):  # pylint: disable=too-many-locals,too-m
 
         upgrade_sstables = self.params.get('upgrade_sstables')
 
-        if num_nodes_to_rollback > 0:
-            # Upgrade all nodes that should be rollback later
-            for i in range(num_nodes_to_rollback):
-                InfoEvent(f'Step{i + 1} - Upgrade node{i + 1}')
-                self.db_cluster.node_to_upgrade = self.db_cluster.nodes[indexes[i]]
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} begins')
-                self.upgrade_node(self.db_cluster.node_to_upgrade, upgrade_sstables=upgrade_sstables)
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} ended')
+        def change_node_version(node_indexes, is_upgrade=True):
+            nonlocal step_idx
+            operation_str = 'Upgrade' if is_upgrade else "Rollback"
+            for random_node_idx in node_indexes:
+                node = self.db_cluster.nodes[random_node_idx]
+                step_idx += 1
+                self.db_cluster.node_to_upgrade = node
+                InfoEvent(f"Step{step_idx} - {operation_str} Node '{node.name}'")
+                self.log.info(f"{operation_str} Node '{node.name}' begins")
+                with ExitStack() as stack:
```

Comment:
please refactor this to a function return a context manager like in the other examples we have

Reply:
Done